pkg/segment/writer/segmetarw.go

```diff
@@ -682,3 +689,90 @@ func writeOverSegMeta(segMetaEntries []*structs.SegMeta) error {
 
 	return nil
 }
+
+func calculateSegmentSizes(segmentKey string) (*SegmentSizeStats, error) {
+	sfm, err := readSfm(segmentKey)
+	if err != nil {
+		return nil, fmt.Errorf("calculateSegmentSizes: failed to read sfm for segment %s: %v", segmentKey, err)
+	}
+
+	stats := &SegmentSizeStats{}
+	for _, colInfo := range sfm.ColumnNames {
+		if colInfo.CmiSize == 0 && colInfo.CsgSize == 0 {
+			continue
+		}
```

Comment:
nit: if these two params are zero, then thats ok, 0 will get added to the below two lines, which is no -op. By removing them we have less code to maintain

Reply:
I used continue here so that stats.NumIndexFiles is not incremented if both CmiSize and CsgSize are zero.